src/doc/rustc/src/platform-support/fuchsia.md

```diff
@@ -0,0 +1,280 @@
+# `aarch64-fuchsia` and `x86_64-fuchsia`
+
+**Tier: 2**
+
+[Fuchsia] is a modern open source operating system that's simple, secure,
+updatable, and performant.
+
+[Fuchsia]: https://fuchsia.dev/
+
+## Target maintainers
+
+The [Fuchsia team]:
+
+[Fuchsia team]: https://team-api.infra.rust-lang.org/v1/teams/fuchsia.json
+
+- Tyler Mandry ([@tmandry](https://github.com/tmandry))
+- Dan Johnson ([@computerdruid](https://github.com/computerdruid))
+- David Koloski ([@djkoloski](https://github.com/djkoloski))
+- Andrew Pollack ([@andrewpollack](https://github.com/andrewpollack))
+- Joseph Ryan ([@P1n3appl3](https://github.com/P1n3appl3))
+
+## Requirements
+
+This target is cross-compiled from a host environment. Development may be done
+from the [source tree] or using the Fuchsia SDK.
+
+[source tree]: https://fuchsia.dev/fuchsia-src/get-started/learn/build
+
+Fuchsia targets support std and follow the `sysv64` calling convention. Fuchsia
+binaries use the ELF file format.
+
+## Building the target
+
+Before building Rust for Fuchsia, you'll need a clang toolchain that supports
+Fuchsia as well. You can download a [prebuilt one] from CIPD Packages or follow
+the guide for [building your own clang toolchain].
+
+[prebuilt one]: https://chrome-infra-packages.appspot.com/p/fuchsia/third_party/clang
+[building your own clang toolchain]: https://fuchsia.dev/fuchsia-src/development/build/toolchain
+
+You'll also need a recent copy of the [Fuchsia SDK], which provides the tools
+and binaries required to build and link programs for Fuchsia.
+
+[Fuchsia SDK]: https://chrome-infra-packages.appspot.com/p/fuchsia/sdk/core
+
+x86-64 and AArch64 Fuchsia targets can be enabled using the following
+configuration.
+
+In `config.toml`, add:
+
+```toml
+[llvm]
+targets = "X86;AArch64"
+
+[build]
+target = ["aarch64-fuchsia", "x86_64-fuchsia"]
+
+[rust]
+llvm-libunwind = "in-tree"
```

Comment:
See https://github.com/rust-lang/rust/pull/93604. This can now be configured per-target like this:

```toml
[target.x86_64-fuchsia]
llvm-libunwind = "in-tree"
[target.aarch64-fuchsia]
llvm-libunwind = "in-tree"
```

Configuring globally works in our builder the way we have it set up for host, but I can't say for sure that it would work in all cases, so it's best to configure per-target like this. (Note: I want to make "in-tree" the default for Fuchsia once we get the sysroot bumped, so this is no longer necessary. cc @andrewpollack)

Reply:
Oh cool, I like that a lot better. Updated.

Reply:
Is this working correctly? there was some confusion about this when @djkoloski was showing this to me today